rust/agents/relayer/src/msg/serial_submitter.rs

```diff
@@ -223,14 +223,17 @@ impl SerialSubmitter {
 
         // Go ahead and attempt processing of message to destination chain.
         debug!(msg=?msg, "Ready to process message");
+        // TODO: consider differentiating types of processing errors, and pushing to the front of the
```

Comment:
Pushing to the front of the run_queue is somewhat dangerous in that you will just keep retrying the same message over and over, and anything behind it in the run_queue will never be attempted.

Reply:
yeah added a comment clarifying that it'd only make sense to do if the # of retries for the message is considered